### DISCUSSION TITLE
Helper to inline SVG icons

### DISCUSSION THREAD

**Discussion by wout** [luckyframework/lucky#1760] [Feature requests & Ideas]
In every one of my Lucky apps I use SVG icons that I style with CSS. They are inlined and sanitized at compile-time, and I use the following snippet:

```crystal
annotation InlineSvg::Path
end
annotation InlineSvg::StripRegex
end

@[InlineSvg::Path("src/icons")]
@[InlineSvg::StripRegex(/(class|fill|stroke|stroke-width|style)="[^"]+" ?/)]
module InlineSvg
  macro svg_icon(path, strip = true)
    {%
      icons_path = InlineSvg.annotation(InlineSvg::Path).args.first
      strip_regex = InlineSvg.annotation(InlineSvg::StripRegex).args.first
      full_path = "#{icons_path.id}/#{path.id}"

      raise "SVG icon #{full_path.id} is missing" unless file_exists?(full_path)

      svg = read_file(full_path).gsub(/<\?xml[^>]+>/, "").gsub(/<!--[^>]+>/, "")
      svg = svg.gsub(strip_regex, "") if strip
    %}

    raw {{svg.gsub(/<svg/, %(<svg data-svg-icon="#{path.id}"))}}
  end
end
```

What this does is:
- inlines SVG files in `src/icons` at compile-time (configurable with an annotation)
- strips the XML declaration and comments
- optionally strips styling attributes so the icons can be styled with CSS
- raises when an icon is missing
- adds a `data-svg-icon` attribute with the path of the icon for icon-specific styling (e.g. `<svg data-svg-icon="social/twitter.svg">`)

It feels silly to create a shard for this. Would It be an idea to include it in the core? I'd be happy to create a PR.

**Comment by jwoertink** [comment]
That looks pretty neat. I'm sure a lot of people these days use SVG Icons. I'd be cool with adding this as a helper.
